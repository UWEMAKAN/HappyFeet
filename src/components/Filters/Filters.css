.Filters {
  display: flex;
  flex-direction: column;
}

.Filters h2 {
  font-size: 2.2rem;
  font-weight: 600;
  margin-top: 3rem;
  margin-bottom: 2rem;
}

.Filters h3 {
  font-size: 1.4rem;
  margin-bottom: 1rem;
  text-transform: uppercase;
}

.Filter {
  padding: 2rem 2rem;
}

@media screen and (max-width: 900px) {
  .Filters h2 {
    font-size: 2rem;
    margin-top: 2rem;
    margin-bottom: 1rem;
  }

  .Filter {
    padding: 2rem 1rem;
  }
}
