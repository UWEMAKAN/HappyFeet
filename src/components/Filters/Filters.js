import React from 'react';
import propTypes from 'prop-types';
import PriceFilter from './PriceFilter/PriceFilter';
import BrandFilter from './BrandFilter/BrandFilter';
import ColorFilter from './ColorFilter/ColorFilter';
import './Filters.css';

const Filters = ({ selectHandler, searchHandler, checkedHandler }) => (
  <div className="Filters">
    <h2>Filters</h2>
    <PriceFilter selectHandler={selectHandler} />
    <BrandFilter searchHandler={searchHandler} />
    <ColorFilter checkedHandler={checkedHandler} />
  </div>
);

Filters.propTypes = {
  selectHandler: propTypes.func.isRequired,
  searchHandler: propTypes.func.isRequired,
  checkedHandler: propTypes.func.isRequired
};

export default Filters;
