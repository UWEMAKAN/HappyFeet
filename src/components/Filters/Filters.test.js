import React from 'react';
import { shallow } from 'enzyme';
import Filters from './Filters';

const renderFilters = (args) => {
  const defaultProps = {
    selectHandler: jest.fn(),
    checkedHandler: jest.fn(),
    searchHandler: jest.fn()
  };
  const props = { ...defaultProps, ...args };
  // eslint-disable-next-line react/jsx-props-no-spreading
  return shallow(<Filters {...props} />);
};

describe('Filters Component', () => {
  test('should render Filters Compponent', () => {
    const wrapper = renderFilters();
    expect.assertions(3);
    expect(wrapper.find('h2').text()).toEqual('Filters');
    expect(wrapper.getElements()[0].props.children.length).toBe(4);
    expect(wrapper).toMatchSnapshot();
  });
});
