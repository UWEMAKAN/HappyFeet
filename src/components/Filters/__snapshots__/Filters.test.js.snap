// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters Component should render Filters Compponent 1`] = `ShallowWrapper {}`;
